Library/Formula/nvm.rb

```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
   sha256 "0008754e7421c55e66f36c4a83ca775928f7b2e7a0543098f0050d9780419b2a"
   head "https://github.com/creationix/nvm.git"
 
+  resource "nvm-exec" do
+    url "https://raw.githubusercontent.com/creationix/nvm/v0.24.1/nvm-exec"
+    sha256 "915118c666c6f3e03ce4a4545038ecae691c7c496579c614f4d32200a180924c"
+  end
+
   def install
     prefix.install "nvm.sh"
     bash_completion.install "bash_completion" => "nvm"
+
+    resource("nvm-exec").stage do
+      prefix.install("nvm-exec")
+      chmod 0755, "#{prefix}/nvm-exec"
+    end
   end
 
   def caveats; <<-EOS.undent
+      Add the following to $HOME/.bashrc, $HOME/.zshrc, or your shell's
+      equivalent configuration file:
+
+        source $(brew --prefix nvm)/nvm.sh
+
+
+      Optionally, if you prefer to have NVM's working directory in your $HOME
+      directory, do the following:
+
       Add NVM's working directory to your $HOME path (if it doesn't exist):
 
         mkdir ~/.nvm
 
-      Add the following to $HOME/.bashrc, $HOME/.zshrc, or your shell's
-      equivalent configuration file:
+      Copy nvm-exec to NVM's working directory
+
+        cp $(brew --prefix nvm)/nvm-exec ~/.nvm/
```

Comment:
As of right now, if you `brew install nvm` you will have a version of `nvm` that is not able to `nvm exec VERSION SCRIPT`.  Trying to do so will get you a cryptic error.

So for a fully functioning `nvm` the user needs to have this script and it needs to wind up in their `NVM_DIR`.

Either homebrew downloads and checks the script or the caveats section should have a line that does that.  I think the value in having homebrew do the checking is the resulting instructions for the user are easier to understand (copy this file vs download & check this file) although I'll admit not by much.

@mikemcquaid  would you prefer the formula to just tell the user how to download and check the script?